File: PoolSpace.py
import requests
import subprocess
import json


endpoint='https://api-mainnet.pool.space/api/farms/'


class PoolSpaceClient():
    def __init__(self, wallet_address):
        self.wallet_address = wallet_address
        self.web_address=(endpoint+wallet_address)
        stdout=subprocess.run(f'curl {self.web_address}',shell=True,capture_output=True).stdout
        self.serialized_data=json.loads(stdout)
        
       

    def requestData(self):
        return self.serialized_data

    def unpaidBalance(self):
        return self.serialized_data["unpaidBalanceInXCH"]
    
    def totalPaidInXCH(self):
        return self.serialized_data["totalPaidInXCH"]

    def blocksFound(self):
        return self.serialized_data["blocksFound"]
    
    def estimatedPlotSizeTiB(self):
        return self.serialized_data["estimatedPlotSizeTiB"]

    def estimatedPlots(self):
        return self.serialized_data["estimatedPlots"]

    def pendingPoints(self):
        return self.serialized_data["pendingPoints"]

    def rank(self):
        return self.serialized_data["rank"]
    

    def printAll(self):
        for i in self.serialized_data:
            print(i+' ==> '+str(self.serialized_data[i]))
    

File: php/PoolSpace.php
<?php
class PoolSpace
{
  // Properties
  public $wallet;
  private $serialized_data;
  private $endpoint;

  function __construct($wallet)
  {
    $this->wallet = $wallet;
    $this->endpoint = 'https://api-mainnet.pool.space/api/farms/' . $wallet . '/';
    $this->serialized_data=json_decode(file_get_contents($this->endpoint));
  }

  // Methods
  function requestData()
  {
    return $this->serialized_data;
  }

  function unpaidBalance()
  {
    return $this->serialized_data->unpaidBalanceInXCH;
  }

  function totalPaidInXCH()
  {
    return $this->serialized_data->totalPaidInXCH;
  }

  function blocksFound()
  {
    return $this->serialized_data->blocksFound;
  }

  function estimatedPlotSizeTiB()
  {
    return $this->serialized_data->estimatedPlotSizeTiB;
  }

  function estimatedPlots()
  {
    return $this->serialized_data->estimatedPlots;
  }

  function pendingPoints()
  {
    return $this->serialized_data->pendingPoints;
  }

  function rank()
  {
    return $this->serialized_data->rank;
  }

  function globalPendingPoints(){
    return $this->serialized_data->globalPendingPoints;
  }

  function printAll(){
      foreach ($this->serialized_data as $key => $value) {
        if (is_array($value))
          print_r($value);
        else
          echo "$key ==> $value \n";
      }
    
    
  }
}